File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<TreeNode*>in;
    void inorder(TreeNode* root)
    {
        if(root== NULL)
        {
            return;
        }

       
        in.push_back(root);
         inorder(root->left);
        inorder(root->right);
    }

   void flatten(TreeNode* root) {

        inorder(root);
        int n=in.size();
        TreeNode* curr=root;
        

        for(int i=1;i<n;i++)
        {
           
            curr->left=NULL;
            curr->right=in[i];

            curr=curr->right;
        }
        

        return ;
    }
};

File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    ArrayList<TreeNode>in=new ArrayList<>();
    public void inorder(TreeNode root)
    {
        if(root==null)
        {
            return;
        }

       
        in.add(root);
        inorder(root.left);
        inorder(root.right);
    }
    public void flatten(TreeNode root) {

         inorder(root);
        int n=in.size();
        TreeNode curr=root;
        

        for(int i=1;i<n;i++)
        {
            curr.left=null;
            curr.right=in.get(i);

            curr=curr.right;
        }
        

        return ;
    }
}